From 19ad270999b978f29365a395d8a72f4a154a2d3b Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 17 Mar 2022 09:05:39 +0000
Subject: [PATCH] GCC: fix name clash of blink::PseudoId

blink::PseudoId is an enum and at the same time a method in
blink::NGHighlightOverlay which returns the enum. GCC errors
out, because it cannot distinguish between both. Add an explicit
enum to the return type to avoid confusion.
---
 .../blink/renderer/core/paint/ng/ng_highlight_overlay.cc        | 2 +-
 third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.cc b/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.cc
index 93f90e2..58da5c1 100644
--- a/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.cc
+++ b/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.cc
@@ -62,7 +62,7 @@ String HighlightLayer::ToString() const {
   return result.ToString();
 }
 
-PseudoId HighlightLayer::PseudoId() const {
+enum PseudoId HighlightLayer::PseudoId() const {
   switch (type) {
     case HighlightLayerType::kOriginating:
       return kPseudoIdNone;
diff --git a/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.h b/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.h
index 42e5f57..44fd214 100644
--- a/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.h
+++ b/third_party/blink/renderer/core/paint/ng/ng_highlight_overlay.h
@@ -40,7 +40,7 @@ class CORE_EXPORT NGHighlightOverlay {
         : type(type), name(name) {}
 
     String ToString() const;
-    PseudoId PseudoId() const;
+    enum PseudoId PseudoId() const;
     const AtomicString& PseudoArgument() const;
 
     int8_t ComparePaintOrder(const HighlightLayer&,
-- 
2.34.1

